P/PETSc/build_tarballs.jl

```diff
@@ -312,5 +322,8 @@ append!(dependencies, platform_dependencies)
 ENV["MPITRAMPOLINE_DELAY_INIT"] = "1"
 
 # Build the tarballs.
+# NOTE: llvm15 seems to have an issue with PETSc 3.18.x as on apple architectures it doesn't know how to create dynamic libraries  
```

Comment:
What's the problem exactly? Also, I think we don't have llvm 15 nor 14 at all, we jumped from v13 to v16, you can see list of compilers in https://github.com/JuliaPackaging/Yggdrasil/tree/29fc7d7beafd210b26fc2c4f3c6cdce76080bfab/0_RootFS

Reply:
Builds for all apple systems were failing with this warning:
```
[20:31:09] *******************************************************************************
[20:31:09]          UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
[20:31:09] -------------------------------------------------------------------------------
[20:31:09] Exhausted all shared linker guesses. Could not determine how to create a shared library!
[20:31:09] *******************************************************************************
```
when using the  `--with-shared-libraries=1` compilation flag. If this flag is not specified (which was the status-quo so far), it compiles a static library instead and, as a result, the audit [fails](https://buildkite.com/julialang/yggdrasil/builds/7414).

That's somewhat puzzling as previous compilations of PETSc_jll (e.g., [here](https://github.com/JuliaPackaging/Yggdrasil/pull/6765)) worked fine. It turns out that these versions use [clang 13.0.1](https://buildkite.com/julialang/yggdrasil/builds/3071#0188387b-a063-4ba0-b7db-f7590cf3b6aa/6-962), whereas the latest build ends up using [clang 16.0.6](https://buildkite.com/julialang/yggdrasil/builds/7414#018cb6bb-fb7f-45f8-a34d-3243c057942c/6-749), and with clang 13 it would create dynamic libraries (even when not specifying the compilation flag).

I expect that there might be a hickup in the PETSc installation system for clang 16 (perhaps that has been resolved in more recent versions of PETSc).
In any case, specifying `preferred_llvm_version=v"14"` results in [clang 13.0.1](https://buildkite.com/julialang/yggdrasil/builds/7437#018cbb9f-9ee0-485b-a8df-6c98d2c222a6/6-953). 

From what you suggests, it should rather be `preferred_llvm_version=v"13"`, not?